lib/ical/recur_iterator.js

```diff
@@ -214,8 +214,19 @@ ICAL.RecurIterator = (function() {
       this.last.second = this.setup_defaults("BYSECOND", "SECONDLY", this.dtstart.second);
       this.last.minute = this.setup_defaults("BYMINUTE", "MINUTELY", this.dtstart.minute);
       this.last.hour = this.setup_defaults("BYHOUR", "HOURLY", this.dtstart.hour);
-      this.last.day = this.setup_defaults("BYMONTHDAY", "DAILY", this.dtstart.day);
+      var day = this.setup_defaults("BYMONTHDAY", "DAILY", this.dtstart.day);
+      // In case of BYMONTHDAY, the day must be set on the correct month otherwise
+      // the automatic normalization might cause a wrong "last" date.
       this.last.month = this.setup_defaults("BYMONTH", "MONTHLY", this.dtstart.month);
+      var daysInMonth = ICAL.Time.daysInMonth(this.last.month, this.last.year);
+      while (Math.abs(day) > daysInMonth) {
```

Comment:
What happens if someone wrongly specifies BYMONTHDAY=32 ? Would this be caught somewhere else or would the loop keep spinning? Maybe you could add a test for this too.